File: shared/backend/noctusai_shared/roles.py
"""
Shared role constants for the NoctusAI platform.

Defines the 7-role org hierarchy used across Core and all products.
Products consume these constants in dependencies.py, routers, and SSO logic.
"""
from __future__ import annotations

# All valid org_role values (noctus_users.org_role)
ORG_ROLES = ("owner", "admin", "manager", "member", "viewer", "dev", "test")

# Roles that grant team/billing management (can invite, remove, change roles)
ADMIN_ROLES = ("owner", "admin")

# Roles that can manage team but NOT billing
MANAGE_TEAM_ROLES = ("owner", "admin", "manager")

# Roles that see "in development" pages (status_pagina.status = 'desenvolvimento')
DEV_ROLES = ("owner", "dev")

# Roles that grant product-level platform_admin via SSO
# (org owner/admin entering a product get full admin access)
PRODUCT_ADMIN_ROLES = ("owner", "admin")

# Portuguese labels for UI display
ORG_ROLE_LABELS = {
    "owner": "Proprietário",
    "admin": "Administrador",
    "manager": "Gerente",
    "member": "Membro",
    "viewer": "Visualizador",
    "dev": "Desenvolvedor",
    "test": "Teste",
}


def is_dev_or_owner(org_role: str | None) -> bool:
    """Check if the user can see in-development pages."""
    return org_role in DEV_ROLES


def can_manage_team(org_role: str | None) -> bool:
    """Check if the user can invite/remove team members."""
    return org_role in MANAGE_TEAM_ROLES


def can_manage_billing(org_role: str | None) -> bool:
    """Check if the user can manage billing/subscription."""
    return org_role in ADMIN_ROLES

File: seed/frontend/lib/src/roles.ts
/**
 * Shared role constants for the NoctusAI platform.
 *
 * Defines the 7-role org hierarchy used across Core and all products.
 * Products consume these in Layout.tsx, admin pages, and invite forms.
 */

export const ORG_ROLES = [
  'owner', 'admin', 'manager', 'member', 'viewer', 'dev', 'test',
] as const;

export type OrgRole = (typeof ORG_ROLES)[number];

/** Roles that grant team/billing management */
export const ADMIN_ROLES: OrgRole[] = ['owner', 'admin'];

/** Roles that can manage team (invite/remove) but not billing */
export const MANAGE_TEAM_ROLES: OrgRole[] = ['owner', 'admin', 'manager'];

/** Roles that see "in development" pages */
export const DEV_ROLES: OrgRole[] = ['owner', 'dev'];

/** Roles that grant product-level platform_admin via SSO */
export const PRODUCT_ADMIN_ROLES: OrgRole[] = ['owner', 'admin'];

/** Portuguese labels for UI display */
export const ORG_ROLE_LABELS: Record<OrgRole, string> = {
  owner: 'Proprietario',
  admin: 'Administrador',
  manager: 'Gerente',
  member: 'Membro',
  viewer: 'Visualizador',
  dev: 'Desenvolvedor',
  test: 'Teste',
};

/** Assignable roles (cannot assign "owner" — that's the org creator only) */
export const ASSIGNABLE_ROLES: OrgRole[] = ['admin', 'manager', 'member', 'viewer', 'dev', 'test'];

/** Check if user can see in-development pages */
export function isDevOrOwner(orgRole: string | null | undefined): boolean {
  return orgRole === 'owner' || orgRole === 'dev';
}

/** Check if user can manage team (invite/remove) */
export function canManageTeam(orgRole: string | null | undefined): boolean {
  return MANAGE_TEAM_ROLES.includes(orgRole as OrgRole);
}

/** Check if user can manage billing */
export function canManageBilling(orgRole: string | null | undefined): boolean {
  return ADMIN_ROLES.includes(orgRole as OrgRole);
}
